### DISCUSSION TITLE
Accessing old app settings

### DISCUSSION THREAD

**Discussion by dkmorb** [dotnet/maui#23813] [Q&A]
Hi all,
I would like to read some user settings from the old Xamarin app. 
In my old application, I used to save settings by calling the method

`PreferenceManager.GetDefaultSharedPreferences(Application.Context)`

Now I have discovered that with Maui it's possible to use the `Preference` class to manage user settings. 

Any ideas on how to access the old settings?

**Comment by phillippschmedt** [comment]
Hi @dkmorb 
while the following documentation is about the version tracking, it has a LegacyPreferences class that shows you how to access the preferences from Xamarin: https://learn.microsoft.com/en-us/dotnet/maui/migration/version-tracking?view=net-maui-8.0

**Reply by dkmorb** [reply]
Hi @phillippschmedt 
I managed to understand and apply the code you pointed out to me yesterday. I modified it to read the settings of another app (the package ids are not the same) using a context created from the name of the old app at runtime. 
But the function ‘GetSharedPreferences’ doesn't seem to work properly because it returns an empty settings map. I am pretty sure that the file name I want to read is correct. 

By the way, here the code I used to experiment:

`var oldContext = context.CreatePackageContext(packageName, PackageContextFlags.IgnoreSecurity);
var settingsList = oldContext.GetSharedPreferences(oldContext.PackageName + "_preferences", FileCreationMode.Private);`

As I said before, settingsList is an empty list ...
